src/lib/crypto/kimchi_bindings/stubs/kimchi_types.ml

```diff
@@ -153,12 +153,40 @@ type nonrec gate_type =
   | ForeignFieldMul
   | Xor16
 
+type nonrec lookup_pattern =
+  | Xor
+  | ChaChaFinal
+  | LookupGate
+  | RangeCheckGate
+  | ForeignFieldMulGate
+
+type nonrec lookup_patterns =
+  { xor : bool
+  ; chacha_final : bool
+  ; lookup_gate : bool
+  ; range_check_gate : bool
+  ; foreign_field_mul_gate : bool
+  }
+[@@boxed]
```

Comment:
Is there a reason for the `boxed` annotation here? (maybe related to underlying Rust bindings?)  AFAIK, OCaml can only unbox single-field records

Reply:
It's something @mimoo added in the ocaml-gen library. We should probably rip it out.

Reply:
Right. I fixed it in https://github.com/o1-labs/ocaml-gen/pull/9/.

Reply:
I remember that it was complaining and I couldn't build without them